### ISSUE TITLE
[BUG] avoid-collection-methods-with-unrelated-types does not compare with dynamic

### ISSUE THREAD

**Issue by praxamarnix** [dart-code-checker/dart-code-metrics#1108]
* **Dart code metrics version:** I use 4.16.0 because of my SDK version. Also verified on 5.2.1.
* **Dart sdk version:** 2.18.0

**Please show your full configuration:**

<details>
<summary>analysis_options.yaml</summary>

```yaml
include: package:lints/recommended.yaml

analyzer:
  plugins:
    - dart_code_metrics

dart_code_metrics:
  rules:
  - avoid-collection-methods-with-unrelated-types
```

</details>

**What did you do? Please include the source code example causing the issue.**

```dart
class Awesome {
  bool foo() {
    dynamic a = 'hello';
    final List<int> b = [1,2,3];
    
    return b.contains(a); // LINT avoid-collection-methods-with-unrelated-types
  }
}
```

**What did you expect to happen?**

I expected an OK, otherwise there is no use for `dynamic`.

**What actually happened?**

linter warning, because a `dynamic` is not an `int`.

**Are you willing to submit a pull request to fix this bug?**

Sure, but I haven't looked in the code yet. I guess I would simply return OK when the type is `dynamic`.

**Comment by incendial** [comment]
Good catch, thank you! I don't see tests for this case, definitely a bug.

**Comment by incendial** [comment]
Ended up keeping current behaviour as the default one, but if you want to allow dynamics or Object types, then the rule should be configured with `strict: false`.

**Comment by incendial** [comment]
Fixed in 5.3.0 🚀
